### PR TITLE
Os board module code

### PR DESCRIPTION
**Pull request by wuqiank**
@Richard
add os_board module code without external library into ossap_stats

### CODE REVIEW

File: ossap_stats/ossap-stats-embed-js.tpl.php [hunk 1]
```diff
@@ -63,7 +75,7 @@
     <?php $value = ($key == 'filesize_bytes') ? format_size($value) : $value ?>
     <?php $value = ($key != "os_version" && is_numeric($value)) ? number_format($value) : $value ?>
     <?php $value = ($key == 'activity-messages') ? $value : "'{$value}'" ?>
-    elements[i].innerHTML = <?php echo $value; ?>;
+	elements[i].innerHTML = <?php echo $value; ?>;
```
Comment (rbran100) [inline ossap_stats/ossap-stats-embed-js.tpl.php]: Looks like you replaced spaces with tabs here, can you set tabs to 2 spaces instead?

File: ossap_stats/ossap_stats.module [hunk 2]
```diff
@@ -31,10 +31,150 @@ function ossap_stats_menu() {
...
+
+/*
+* _os_board_get_require_once()
+* no parameters. all the required files in this function
+* 
+*/
+function _os_board_get_require_once(){
+  require_once 'google-api-php-client/src/Google_Client.php';
+  require_once 'google-api-php-client/src/contrib/Google_AnalyticsService.php';
+  require_once 'os_board.inc';
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module:130]: Will this file exist?

File: ossap_stats/ossap_stats.module [hunk 3]
```diff
@@ -31,10 +31,150 @@ function ossap_stats_menu() {
     'access callback' => TRUE,
     'type' => MENU_CALLBACK,
   );
-
+  //osboard menu start
+  $items['ossap/most-viewed-pages'] = array(
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module:35]: Not sure you still need this menu entry?

File: ossap_stats/ossap_stats.module [hunk 4]
```diff
@@ -31,10 +31,150 @@ function ossap_stats_menu() {
...
+    'title' => t('OpenScholar Board - Most Viewed Pages'),
+	'type' => MENU_CALLBACK,
+    'page callback' => '_os_board_viewed_pages',
+	'access callback' => TRUE,
+  );
+  
+  $items['ossap/most-visited-sites'] = array(
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module:42]: or this one?

File: ossap_stats/ossap_stats.module [hunk 5]
```diff
@@ -46,6 +186,9 @@ function ossap_stats_theme() {
         'os_version' => NULL,
         'messages' => NULL,
         'getsatisfaction' => NULL,
+		'os_board_visited_pages' => NULL,
+		'os_board_viewed_pages' => NULL
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module]: tabs again

File: ossap_stats/ossap_stats.module [hunk 6]
```diff
@@ -120,6 +276,9 @@ function _ossap_stats_embed_js($stat) {
     'os_version' => _ossap_stats_get_os_version(),
     'messages' => $messages,
     'getsatisfaction' => $getsatisfaction,
+	//added variables for most_visited_sites and most_viewed_pages
+	'most_visited_sites' => $most_visited_sites,
+	'most_viewed_pages' => $most_viewed_pages
```
Comment (rbran100) [inline ossap_stats/ossap_stats.module]: tabs again, I will stop mentioning :-)

File: ossap_stats/ossap_stats_credentials.inc [hunk 7]
```diff
@@ -0,0 +1,127 @@
+<?php
+
+/* authenticate credentials by creating a new Google_client object and 
+* sending it to google analytics
+* check_credentials
+* no parameters
+* creates the google_client object
+* returns the validated service object
+*/
+function check_credentials(){
```
Comment (rbran100) [inline ossap_stats/ossap_stats_credentials.inc]: You need to namespace your function in here ex:

```
ossap_stats_check_credentials();
```

File: ossap_stats/ossap_stats.module (hunk 2 above)
Comment (rbran100) [inline ossap_stats/ossap_stats.module:130]: You can use: module_load_include()

**Comment by priya-sudarshanam** [comment]
@Richard
add os_board module code without external library into ossap_stats

A little confused by the above:
Can you pls further elaborate on what it means

Thanks,
Priya

**Comment by rbran100** [comment]
@priya-sudarshanam I was saying that you can use https://api.drupal.org/api/drupal/includes!module.inc/function/module_load_include/7 module_load_include instead of require once

Also I didn't see os_board.inc in the Filesystem, are we sure that file exists?

**Comment by wuqiank** [comment]
@rbran100 
We have replaced "require once" with "module_load_include" 
and we changed the .inc file name, "os_board.inc" was a wrong name of our existing .inc, so only one .inc file

Thanks a lot

**Comment by priya-sudarshanam** [comment]
Hi Richard,
   Your instructions for the module_load_include was clear but I was confused on the following part:
"do you mind separating the external library and the module code into two separate PR's".

Yes you are right regarding the os_board.inc, it does not exist. It has now been renamed to ossap_stats_credentials.inc as Qi mentioned. The module_load_include
 is in and the function check_credentials has now been namespaced.

Thanks,
Priya

From: Richard Brandon [notifications@github.com]Sent: Thursday, May 01, 2014 5:00 PMTo: openscholar/ossapCc: Sudarshanam, PriyashreeSubject: Re: [ossap] Os board module code (#23)

@priya-sudarshanam I was saying that you can use

https://api.drupal.org/api/drupal/includes!module.inc/function/module_load_include/7 module_load_include instead of require once
Also I didn't see os_board.inc in the Filesystem, are we sure that file exists?
—
Reply to this email directly or 
view it on GitHub.

**Pull request merged**